src/page/Login/hooks/useLogin.ts

```diff
@@ -1,33 +1,34 @@
 import { requests } from '@/api/requests';
 import { useUserInfo } from '@/contexts/UserInfoContext';
+import { ROUTE_PATH } from '@/routes/routePath';
+import type { UserInfoData, UserInfoProps } from '@/types';
+import { useMutation } from '@tanstack/react-query';
+import { useLocation, useNavigate } from 'react-router-dom';
 import { toast } from 'react-toastify';
 
-export interface UserInfoProps {
-  username: { value: string };
-  password: { value: string };
-}
-
-export interface UserInfoData {
-  name: string;
-  email: string;
-  authToken: string;
-}
-
 const useLogin = () => {
   const { setLoginSession } = useUserInfo();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const from = location.state?.from?.pathname || ROUTE_PATH.MY;
 
-  const loginAndStoreSession = async ({ username, password }: UserInfoProps): Promise<boolean> => {
-    try {
-      const userInfoData = await requests.fetchUserInfos({ username, password });
-      setLoginSession(userInfoData);
-      toast(userInfoData.email);
-      return true;
-    } catch {
-      return false;
-    }
-  };
+  const { mutate } = useMutation<UserInfoData, Error, UserInfoProps>({
+    mutationFn: requests.fetchUserInfos,
+    onSuccess: (data: UserInfoData) => {
+      setLoginSession(data);
+      toast(`${data.name}님, 환영합니다!`);
+      navigate(from, { replace: true });
+    },
+    onError: (error: Error) => {
+      if (error instanceof Error) {
+        toast.error(error.message);
+      } else {
+        toast.error('알 수 없는 오류가 발생했습니다.');
+      }
+    },
+  });
 
-  return { loginAndStoreSession };
+  return { login: mutate };
```

Comment:
```suggestion
  return mutate
```
그냥 mutate를 반환해버리면 어떨까요?

그럼 이렇게 사용할 수 있겠죠?
```tsx
const login = useLogin();
```

그리고 바깥에서 이름을 지정하기도 편해진답니다 ㅎㅎ